src/modules/hyprland/workspaces.cpp

```diff
@@ -56,55 +68,73 @@ auto Workspaces::update() -> void {
     create_workspace(workspace_to_create);
   }
 
-  workspaces_to_create_.clear();
-
-  for (std::unique_ptr<Workspace> &workspace : workspaces_) {
-    workspace->set_active(workspace->id() == active_workspace_id);
-
+  for (auto &workspace : workspaces_) {
+    workspace->set_active(workspace->name() == active_workspace_name);
     std::string &workspace_icon = icons_map_[""];
     if (with_icon_) {
       workspace_icon = workspace->select_icon(icons_map_);
     }
-
     workspace->update(format_, workspace_icon);
   }
-
   AModule::update();
 }
 
 void Workspaces::onEvent(const std::string &ev) {
   std::lock_guard<std::mutex> lock(mutex_);
   std::string eventName(begin(ev), begin(ev) + ev.find_first_of('>'));
   std::string payload = ev.substr(eventName.size() + 2);
+
   if (eventName == "workspace") {
-    std::from_chars(payload.data(), payload.data() + payload.size(), active_workspace_id);
+    active_workspace_name = payload;
+
   } else if (eventName == "destroyworkspace") {
-    int deleted_workspace_id;
-    std::from_chars(payload.data(), payload.data() + payload.size(), deleted_workspace_id);
-    workspaces_to_remove_.push_back(deleted_workspace_id);
+    workspaces_to_remove_.push_back(payload);
+
   } else if (eventName == "createworkspace") {
-    int new_workspace_id;
-    std::from_chars(payload.data(), payload.data() + payload.size(), new_workspace_id);
-    workspaces_to_create_.push_back(new_workspace_id);
+    const Json::Value workspaces_json = gIPC->getSocket1JsonReply("workspaces");
+    for (auto &workspace_json : workspaces_json) {
+      if (workspace_json["name"].asString() == payload &&
+          (all_outputs() || bar_.output->name == workspace_json["monitor"].asString()) &&
+          (workspace_json["name"].asString().find("special:") != 0 || show_special()))
```

Comment:
Can you please explain what special workspace is?
Didn't found any info in Hyprland wiki.